TrustStoreResource does not work on Java 11

Another case that has been masked by #723.

There is no `jre/lib/security/cacerts` with JDKs >= 10 and `GRAALVM_HOME` is not set in the CI environment, hence we see:

```
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running org.apache.camel.quarkus.component.http.it.HttpTest
[ERROR] Tests run: 8, Failures: 0, Errors: 1, Skipped: 7, Time elapsed: 0.245 s <<< FAILURE! - in org.apache.camel.quarkus.component.http.it.HttpTest
[ERROR] org.apache.camel.quarkus.component.http.it.HttpTest.httpsProducer(String)[1]  Time elapsed: 0.01 s  <<< ERROR!
java.lang.RuntimeException: java.lang.RuntimeException: Unable to start Quarkus test resource org.apache.camel.quarkus.test.TrustStoreResource@7068e664
Caused by: java.lang.RuntimeException: Unable to start Quarkus test resource org.apache.camel.quarkus.test.TrustStoreResource@7068e664
Caused by: java.lang.IllegalStateException: Could not find any existing file to set javax.net.ssl.trustStore; tried $GRAALVM_HOME/jre/lib/security/cacerts and $JAVA_HOME/jre/lib/security/cacerts. You may need to set GRAALVM_HOME or JAVA_HOME properly. Found $GRAALVM_HOME = null and $JAVA_HOME = null
```

--- a/integration-tests/http/src/test/java/org/apache/camel/quarkus/component/http/it/HttpIT.java
+++ b/integration-tests/http/src/test/java/org/apache/camel/quarkus/component/http/it/HttpIT.java
@@ -17,11 +17,7 @@
 package org.apache.camel.quarkus.component.http.it;
 
 import io.quarkus.test.junit.NativeImageTest;
-import org.junit.jupiter.api.condition.EnabledOnJre;
-import org.junit.jupiter.api.condition.JRE;
 
 @NativeImageTest
-// TrustStoreResource does not work on Java 11: https://github.com/apache/camel-quarkus/issues/852
-@EnabledOnJre(JRE.JAVA_8)
 class HttpIT extends HttpTest {
 }

--- a/integration-tests/http/src/test/java/org/apache/camel/quarkus/component/http/it/HttpTest.java
+++ b/integration-tests/http/src/test/java/org/apache/camel/quarkus/component/http/it/HttpTest.java
@@ -21,8 +21,6 @@ import io.quarkus.test.junit.QuarkusTest;
 import io.restassured.RestAssured;
 import org.apache.camel.quarkus.test.TrustStoreResource;
 import org.junit.jupiter.api.Test;
-import org.junit.jupiter.api.condition.EnabledOnJre;
-import org.junit.jupiter.api.condition.JRE;
 import org.junit.jupiter.params.ParameterizedTest;
 import org.junit.jupiter.params.provider.ValueSource;
 
@@ -32,8 +30,6 @@ import static org.hamcrest.Matchers.is;
 @QuarkusTest
 @QuarkusTestResource(HttpTestResource.class)
 @QuarkusTestResource(TrustStoreResource.class)
-// TrustStoreResource does not work on Java 11: https://github.com/apache/camel-quarkus/issues/852
-@EnabledOnJre(JRE.JAVA_8)
 class HttpTest {
     @ParameterizedTest
     @ValueSource(strings = { "ahc", "http", "netty-http" })

--- a/integration-tests/support/test-support/src/main/java/org/apache/camel/quarkus/test/TrustStoreResource.java
+++ b/integration-tests/support/test-support/src/main/java/org/apache/camel/quarkus/test/TrustStoreResource.java
@@ -24,23 +24,30 @@ import java.util.Map;
 
 import io.quarkus.test.common.QuarkusTestResourceLifecycleManager;
 
+import static org.apache.camel.util.ObjectHelper.isNotEmpty;
+
 /**
  * Native images built using a docker container end up with {@code javax.net.ssl.trustStore} system property
  * pointing to a non-existing file; see https://quarkus.io/guides/native-and-ssl For that case, we have to set
  * {@code javax.net.ssl.trustStore} to an existing path explicitly.
  */
 public class TrustStoreResource implements QuarkusTestResourceLifecycleManager {
+    private static final String CACERTS_REL_PATH = "lib/security/cacerts";
+    private static final String CACERTS_REL_PATH_ALT = "jre/lib/security/cacerts";
 
     @Override
     public Map<String, String> start() {
         final String graalVmHome = System.getenv("GRAALVM_HOME");
         final String javaHome = System.getenv("JAVA_HOME");
-        Path trustStorePath = null;
-        final String CACERTS_REL_PATH = "jre/lib/security/cacerts";
-        if (graalVmHome != null && !graalVmHome.isEmpty()
-                && Files.exists(trustStorePath = Paths.get(graalVmHome).resolve(CACERTS_REL_PATH))) {
-        } else if (javaHome != null && !javaHome.isEmpty()
-                && Files.exists(trustStorePath = Paths.get(javaHome).resolve(CACERTS_REL_PATH))) {
+
+        Path trustStorePath;
+
+        if (isNotEmpty(graalVmHome) && Files.exists(trustStorePath = Paths.get(graalVmHome).resolve(CACERTS_REL_PATH))) {
+            // empty body
+        } else if (isNotEmpty(javaHome) && Files.exists(trustStorePath = Paths.get(javaHome).resolve(CACERTS_REL_PATH))) {
+            // empty body
+        } else if (isNotEmpty(javaHome) && Files.exists(trustStorePath = Paths.get(javaHome).resolve(CACERTS_REL_PATH_ALT))) {
+            // empty body
         } else {
             throw new IllegalStateException(
                     "Could not find any existing file to set javax.net.ssl.trustStore; tried $GRAALVM_HOME/" + CACERTS_REL_PATH
@@ -48,11 +55,11 @@ public class TrustStoreResource implements QuarkusTestResourceLifecycleManager {
                             + ". You may need to set GRAALVM_HOME or JAVA_HOME properly. Found $GRAALVM_HOME = " + graalVmHome
                             + " and $JAVA_HOME = " + graalVmHome);
         }
+
         return Collections.singletonMap("javax.net.ssl.trustStore", trustStorePath.toString());
     }
 
     @Override
     public void stop() {
     }
-
 }

